### PR TITLE
Find and hook NianticTrustManager methods by signature

### PR DESCRIPTION
**Pull request by trisk**
Haven't had a chance to test this yet, but this is here in case they decide to start obfuscating method names.

### CODE REVIEW

**Comment by trisk** [comment]
I was able to confirm that this works. It hooks both `checkServerTrusted` and `checkClientTrusted` (since it can't tell them apart without relying on the names). As you may have noticed, both of them wrap `nativeCheckServerTrusted` -- for some reason, there are no callers of  `nativeCheckClientTrusted`.

Additionally, forcing both methods to return without calling the native method appears to be safe (arriving at the same state as a successful compare) in the current implementation, based on partial disassembly of NianticLabsPlugin. Will also look into `getAcceptedIssuers`.

**Comment by rastapasta** [comment]
@trisk, i realllly like your passion and skills! thanks again for wrapping your mind around it and this beautiful idea for searching for the method to hook!

I'm just wondering if it's really necessary, as the TrustManager is just [implementing](https://gist.github.com/anonymous/25937fede6b0984d44edfc9f2a9305de#file-niantictrustmanager-class-L14) the [X509TrustManager](https://docs.oracle.com/javase/7/docs/api/javax/net/ssl/X509TrustManager.html) to override some methods. So renaming them would get tricky.
What they could rename though is the NianticTrustManager itself :)

As much as I love your approach, I would like to keep it simple for now but keep it in the archive for the (surely coming) time when we have to adapt to their changes.

Also thanks for taking a closer look at the .so!
What you found out about the safeness about just never calling the nativeCheckServerTrusted is highly interesting - expected them to have some validation / blocking implemented, but they might just never expected it to no being called 😄 

So basically overriding the checkServerTrusted with an empty method would do the trick as well?

Still sticking a bit to the way of injecting the original chain of trust to keep the communication between their native lib and the system as untouched as possible (in terms of all expected calls to happen, even though your research suggests its currently irrelevant).

Thanks for all your energy!

**Comment by trisk** [comment]
Hah, yeah I hadn't considered that they probably can't change the method names baked into the interface. 😄 Let's close this for now.
